src/Microsoft.TemplateEngine.Core/Conditional.cs

```diff
@@ -2,95 +2,191 @@
+
+        // The other operations to toggle when the actionable tokens get handled
+        // They're disabled during initial setup of conditional.
+        // When the actionable token start is encountered, these are enabled.
+        // When the actionable token end is encountered, they're disabled.
+        //  it gets a bit more complex with embedded actionables.
+        private readonly IList<string> _actionableOperations;
 
         public bool WholeLine => _wholeLine;
 
         public bool TrimWhitespace => _trimWhitespace;
 
         public ConditionEvaluator Evaluator => _evaluator;
 
-        public Conditional(string ifToken, string elseToken, string elseIfToken, string endIfToken, bool wholeLine, bool trimWhitespace, ConditionEvaluator evaluator)
+        // standard versions of the tokens.
+        public IList<string> IfTokens => _ifTokens;
+
+        public IList<string> ElseIfTokens => _elseIfTokens;
+
+        public IList<string> ElseTokens => _elseTokens;
+
+        public IList<string> EndIfTokens => _endIfTokens;
+
+        // for comment / uncomment handling
+        public IList<string> IfTokensActionable => _ifTokensActionable;
+
+        public IList<string> ElseTokensActionable => _elseTokensActionable;
+
+        public IList<string> ElseIfTokensActionable => _elseIfTokensActionable;
+
+        private string _id;
+
+
+        public Conditional(ConditionalTokens tokenVariants, bool wholeLine, bool trimWhitespace, ConditionEvaluator evaluator, string id = null)
         {
             _trimWhitespace = trimWhitespace;
             _wholeLine = wholeLine;
             _evaluator = evaluator;
-            _ifToken = ifToken;
-            _elseToken = elseToken;
-            _elseIfToken = elseIfToken;
-            _endIfToken = endIfToken;
+
+            _ifTokens = tokenVariants.IfTokens;
+            _elseTokens = tokenVariants.ElseTokens;
+            _elseIfTokens = tokenVariants.ElseIfTokens;
+            _endIfTokens = tokenVariants.EndIfTokens;
+            _ifTokensActionable = tokenVariants.ActionableIfTokens;
+            _elseTokensActionable = tokenVariants.ActionableElseTokens;
+            _elseIfTokensActionable = tokenVariants.ActionableElseIfTokens;
+
+            _actionableOperations = tokenVariants.ActionableOperations;
+            _id = id;
         }
 
-        public IOperation GetOperation(Encoding encoding, IProcessorState processorState)
+        /// <summary>
+        /// Returns the numner of elements in the longest of the token variant lists.
+        /// </summary>
+        private int LongestTokenVariantListSize
         {
-            byte[] ifToken = encoding.GetBytes(_ifToken);
-            byte[] endToken = encoding.GetBytes(_endIfToken);
-
-            List<byte[]> tokens = new List<byte[]>
+            get
             {
-                ifToken,
-                endToken
-            };
+                int maxListSize = Math.Max(_ifTokens.Count, _elseTokens.Count);
```

Comment:
IIRC there's an overload of Math.Max that accepts an arbitrary number of values to take the max of